lib/rules/no-incorrect-calls-with-inline-anonymous-functions.js

```diff
@@ -0,0 +1,83 @@
+'use strict';
+
+const utils = require('../utils/utils');
+const ERROR_MESSAGE =
+  'Do not use anonymous functions as arguments to `debounce`, `once`, and `scheduleOnce`.';
+
+function isMemberExpressionOnRun(node) {
+  return utils.isMemberExpression(node.callee) && node.callee.object.name === 'run';
+}
+
+const functionRules = [
+  { importPath: '@ember/runloop', importName: 'debounce' },
+  { importPath: '@ember/runloop', importName: 'once' },
+  { importPath: '@ember/runloop', importName: 'scheduleOnce' },
+];
+
+const allDedupingRunMethodNames = functionRules.map(rule => rule.importName);
+
+module.exports = {
+  ERROR_MESSAGE,
+
+  meta: {
+    docs: {
+      description:
+        'Disallows inline anonymous functions as arguments to `debounce`, `once`, and `scheduleOnce`',
+      category: 'Possible Errors',
+      recommended: false,
+    },
+  },
+
+  create(context) {
+    function checkArgumentsForInlineFunction(node) {
+      node.arguments.forEach((argument, index) => {
+        if (utils.isAnyFunctionExpression(argument)) {
+          context.report(node.arguments[index], ERROR_MESSAGE);
+        }
+      });
+    }
+    function checkFunctionRule(functionRule, node) {
+      if (functionRule.importName === node.callee.name) {
+        checkArgumentsForInlineFunction(node);
+      }
+    }
+
+    let importedRun = false;
+    let inactiveFunctionRules = new Set(functionRules);
+    let activeFunctionRules = new Set();
+
+    return {
+      ImportDeclaration(node) {
+        const importPath = node.source.value;
+        const namedImports = node.specifiers.map(specifier => {
+          return specifier.imported.name;
+        });
+
+        [...inactiveFunctionRules].forEach(functionRule => {
+          if (
+            functionRule.importPath === importPath &&
+            namedImports.includes(functionRule.importName)
+          ) {
+            inactiveFunctionRules.delete(functionRule);
+            activeFunctionRules.add(functionRule);
+          }
+        });
+
+        if (node.source.value === '@ember/runloop' && namedImports.includes('run')) {
```

Comment:
I thought the rule was supposed to depend fully on the `functionRules` list to know about the imports/functions it applies to.

Reply:
It wouldn't share any logic with the other rules, so it felt off to put it in `functionRules`. Abstracting it without a second case where we have an imported object with chained function calls feels premature. 

It won't come in useful for `addEventListener` because they're not called on an imported object. `computed` is the only other named import that has a similar chaining style to `run` that jumped to mind. I think this is a fairly unique case.